xdsl/transforms/memref_to_dsd.py

```diff
@@ -119,7 +123,48 @@ class LowerSubviewOpPass(RewritePattern):
 
     @op_type_rewrite_pattern
     def match_and_rewrite(self, op: memref.SubviewOp, rewriter: PatternRewriter, /):
-        assert isa(op.source.type, MemRefType[Attribute])
+        assert isa(op.source.type, AnyMemRefType)
+        assert isa(op.result.type, AnyMemRefType)
+
+        if len(op.result.type.get_shape()) == 1 and len(op.source.type.get_shape()) > 1:
+            # 1d subview onto a nd memref
+            sizes = op.static_sizes.get_values()
+            counter_sizes = collections.Counter(sizes)
+            if 1 in counter_sizes:
+                counter_sizes.pop(1)
```

Comment:
You can do `counter_sizes.pop(1, None)` to remove `1` if it's present and do nothing otherwise. Though this may be less readable

Reply:
Good point!